pyccel/ast/operators.py

```diff
@@ -1070,7 +1070,91 @@ class PyccelIsNot(PyccelIs):
     def __repr__(self):
         return '{} is not {}'.format(self.args[0], self.args[1])
 
+
 #==============================================================================
 
+class IfTernaryOperator(Basic, PyccelOperator):
+    """Represent a ternary conditional operator in the code, of the form (a if cond else b)
+
+    Parameters
+    ----------
+    args :
+        args : type list
+        format : condition , value_if_true, value_if_false
+
+    Examples
+    --------
+    >>> from sympy import Symbol
+    >>> from pyccel.ast.core import Assign
+	>>>	from pyccel.ast.operators import IfTernaryOperator
+    >>> n = Symbol('n')
+    >>> x = 5 if n > 1 else 2
+    >>> IfTernaryOperator(PyccelGt(n > 1),  5,  2)
+    IfTernaryOperator(PyccelGt(n > 1),  5,  2)
+    """
+    _precedence = 3
+    def __init__(self, cond, value_true, value_false):
+        PyccelOperator.__init__(self, cond, value_true, value_false)
+        self._cond = self._args[0]
+        self._value_true = self._args[1]
+        self._value_false = self._args[2]
+        if self.stage == 'syntactic':
+            return
+        if isinstance(value_true , Nil) or isinstance(value_false, Nil):
+            errors.report('None is not implemented for Ternary Operator', severity='fatal')
+        if isinstance(value_true.dtype, NativeString) or isinstance(value_false.dtype, NativeString):
+            errors.report('Strings are not supported by Ternary Operator', severity='fatal')
+        _tmp_list = [NativeBool(), NativeInteger(), NativeReal(), NativeComplex(), NativeString()]
+        if value_true.dtype not in _tmp_list :
```

Comment:
Is this list needed or could it be simplified to `if not isinstance (value_true.dtype, DataType)` ? (@MenoIy )

Reply:
the list is needed for getting the data type of the ternary operator.

Reply:
Ah yes I see. So then it is needed in the `_set_dtype` function but not here?

Reply:
@MenoIy ?

Reply:
@mostafaMamoni maybe you can try this? I can't think of any reason why it wouldn't work

Reply:
is there any function set_type? like the one in PyccelBinaryOperator ?

Reply:
it should be ok using set_dtype  now that it inherits from pycelOperator